src/Components/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { DropdownProps, DropdownOption } from "./types";
+import { Autocomplete, TextField, Box, InputAdornment } from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search"; // Import the search icon
+import { DropDownType } from "./consts";
+
+const DropDown = ({
+  type = DropDownType.REGULAR,
+  options,
+  value,
+  onChange,
+  label = "",
+  style,
+}: DropdownProps) => {
+  const [showSearchIcon, setShowSearchIcon] = useState(
+    type == DropDownType.SEARCH
+  );
+  const handleChange = (event: any, newValue: DropdownOption | null) => {
+    setShowSearchIcon(false);
+    if (onChange && newValue) {
+      onChange(newValue.value);
+    } else if (!newValue) {
+      onChange("");
+      setShowSearchIcon(type == DropDownType.SEARCH);
+    }
+  };
```

Comment:
pay attention to:
1. onChange is required and not optional, therefore this check is unnecessary and confusing.
2. you are not using the event argument being oasses to the function (and it makes sense, since you are passing the newValue).

change to:
```
  const handleChange = (newValue: DropdownOption | null) => {
    setShowSearchIcon(false);
    if (newValue) {
      onChange(newValue.value);
    } else  {
      onChange("");
      setShowSearchIcon(type == DropDownType.SEARCH);
    }
  };
```

Reply:
another thing: 
why do you need this:
`setShowSearchIcon(type == DropDownType.SEARCH);`